src/main/java/httpserver/MethodWrapper.java

```diff
@@ -1,14 +1,16 @@
 package httpserver;
 
-import java.lang.reflect.Array;
-import java.lang.reflect.Constructor;
-import java.lang.reflect.InvocationTargetException;
-import java.lang.reflect.Method;
 import java.math.BigDecimal;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
+import java.util.HashMap;
 
+// TODO Rewrite almost all of the javadocs for this class.
+//      They're based on old, reflective, less conventional ways of doing
+//      things.
+//      Also, just clean this file up in general.
+//      And maybe rename it to "RouteWrapper?"

```

Comment:
Is there any reason _not_ to just merge this with `Route`?

As in, make `Route` abstract (with the only abstract thing being a `handle` method), and have it basically do all of this?

So instead of writing

```java
server.get("/" new Route() { /* override handle */ });
```

You'd write

```java
server.get(new Route("/") { /* override handle */ });
```